### ISSUE TITLE
Sentry Upgrade and Clean Up

### ISSUE THREAD

**Issue by johnpaulashenfelter** [department-of-veterans-affairs/va.gov-team#6769]
## Background

Sentry released a major upgrade of [on-premise](https://blog.sentry.io/2020/01/07/self-hosted-sentry-10-is-ready-to-serve-get-it-while-its-hot) offering in January 2020. Our installation can be upgraded but this will require some methodical discovery and planning. 

While we could feasibly upgrade our existing instances to the next version, we will take this opportunity to bring Sentry up to speed with the VSP containerized infrastructure approach, leveraging Docker and Terraform to configure and orchestrate updated infrastructure.

## High Level User Story/ies

- As a VSP engineer I can run Sentry and dependencies with `docker-compose` so that I spend less time configuring and more time iterating
- As a VSP engineer I have access to Sentry 10.0 so I can leverage new features and performance

## Hypothesis or Bet

If we upgrade our on-premise Sentry instance, we expect VSP users to...

- ... have a  clear, documented methodology for deploying new Dockerized utilities
- ... feel comfortable deploying new utilities to VSP utilizing best practices 
- ... reduce the overall toil required to upgrade on-premise Sentry utility in the future 

## OKR

Specifically, this effort will help to reach the following results:

- Reduction of manual support requests by leveraging new integration features **(O1)**
- Stability of the platform's systems will improve by modernizing tools, and normalizing upgrade path **(O3)**
- Teams can automatically enforce error handling protocols by leveraging new Sentry integrations w/ Slack & GitHub **(O5)**

### What must be true in order for you to consider this epic complete?

- [ ] on-premise Sentry version is at `10.0.x`
- [ ] local `docker-compose` based workflow is developed and documented
- [ ] all appropriate specs and documentation are provided
- [ ] utility environment deploy process is documented
- [ ] operations team has approved changes
- [ ] errors are populating correctly in upgraded version

## How to configure this issue
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `tools-be`, `tools-fe`,`content-ia`, `service-design`, `vsa-*`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)

**Comment by johnpaulashenfelter** [comment]
One thing yall might consider breaking out into its own issue is a deployment plan. I really don't know if this is a "do staging, verify, then production" or a "run in parallel while we make sure its ok" or "flip the switch" or some other pattern. Considering all the moving pieces, I'm not sure its a direct swap, though that's what you can figure out :)

Also worth thinking about if its ok just to nuke all the data and start fresh...

**Comment by kfrz** [comment]
@johnpaulashenfelter Definitely an upfront deliberation point -> I'm leaning towards running in parallel (starting fresh), but want to get some specifics mapped out first. We don't use Sentry to inform any automated alerting or reporting, AFAIK we'd be losing historical context for folks. 

That said if we run in parallel it's akin to a fresh db anyhow.

**Comment by LindseySaari** [comment]
[Self hosting sentry with docker](https://mikedombrowski.com/2018/03/self-hosting-sentry-with-docker-and-docker-compose/)

**Comment by annaswims** [comment]
The Benefits-memorial team has this ticket and I think it would be a good opportunity for us to work on with them to understand needs https://github.com/department-of-veterans-affairs/va.gov-team/issues/6813

**Comment by LindseySaari** [comment]
## Our thoughts: 
* **First:** upgrade to 9.1.2 — make sure we have a DB backup just incase
* Going to 10 isn’t going to be so simple, a lot of new stuff (TF changes, new deployment config (brand new), writing some new playbooks)
* What’s the benefit of upgrading to 10? -- Research and add to RFC
* Running v9 and 10 in parallel: Sentry 10 is going to be a new deployment before it’s going to be an upgrade
* Pros/Cons of hosted Sentry (ATO?) — Revisit this idea and add to RFC

### Plan moving forward:
1. Create Architecture Doc for Sentry 10 
2. Create RFC for Architecture - https://github.com/department-of-veterans-affairs/va.gov-team/pull/7181
3. Get sentry 10 running locally 
4. Research the benefit of Sentry 10
5. Understand all moving parts and dependencies

**Comment by LindseySaari** [comment]
[Sentry 10 converstation in #vsp-operations](https://dsva.slack.com/archives/CJYRZK2HH/p1585063979460800)
